Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java

```diff
@@ -925,11 +926,69 @@ public void onChannelSelected(String channelName) {
 
     //----------App Bridge---------------//
 
-    public void listOnLocalIndex() {
-        Branch.getInstance().addToSharedContent(this);
+    /**
+     * <p>
+     * Makes the content associated with this BUO discoverable in the native search on Samsung devices.
+     * </p>
+     * <p/>
+     * <p>
+     * Calling this method would put the BUO to the Samsung local search. Your content will appear in
+     * Samsung search as user search with matching keywords. Make sure the BUO is properly configured with
+     * {@link #setCanonicalIdentifier(String)},{@link #setTitle(String)}, {@link #setContentDescription(String)}, {@link #setContentImageUrl(String)} etc.
+     * The Title, Description and Image url is used to show your content properly in the search results.
+     * The contents displayed in Samsung search will get directly deep linked to the app.
+     * </p>
+     *
+     * @param userAction A {@link String }with value of user action name. See {@link io.branch.referral.util.BranchEvent} for Branch defined user events.
+     * @return {@code true} if content successfully added to samsung search
+     */
+    public boolean listOnSamsungSearch(String userAction) {
+        boolean isContentAdded;
+        Context context = Branch.getInstance().getAppContext();
+        String url = getShortUrl(context, new LinkProperties().setChannel("Branch Search"));
+        isContentAdded = BranchSearchServiceConnection.getInstance().addToSharableContent(this, context.getPackageName(), url);
+        if (!TextUtils.isEmpty(userAction)) {
```

Comment:
These are the service connection methods and service connection make call to aidl. Agree that it is good to have same names